src/NServiceBus.Core/Pipeline/Incoming/IncomingPhysicalMessageContext.cs

```diff
@@ -15,7 +15,7 @@ public IncomingPhysicalMessageContext(IncomingMessage message, IBehaviorContext
 
         public void UpdateMessage(byte[] body)
         {
-            Message.Body = body;
+            Message.UpdateBody(body);
```

Comment:
looking at the usages of `UpdateBody` wouldn't make sense to go a step further and make it public while removing the `UpdateMessage` method from the context? Or maybe it still makes sense to limit the stages where you can actually update the body?